utils/backoff_test.go

```diff
@@ -0,0 +1,64 @@
+package utils
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func TestProgressiveRetryShouldFail(t *testing.T) {
+	i := 0
+	operationError := errors.New("Operation Failed")
+	retryOperation := func() error {
+		i++
+		return operationError
+	}
+
+	err := ProgressiveRetry(retryOperation)
+
+	if err == nil || err != operationError {
+		t.Errorf("Expected %v, Got %v", operationError, err)
```

Comment:
We've taken to using https://godoc.org/github.com/stretchr/testify/assert instead of just the built in Go testing primitives, as they add significant clarity when a test fails. Can you rewrite using same? (e.g. `assert.EqualError(err, ...)`

Reply:
There's also a `...require` version that interrupts the flow of execution, useful for checking errors for instance.

OT: anyone knows of a unit test generator that uses testify? I've been toying with unit test generator built into VSCode, it's not bad even if the tests are ++complicated. Unfortunately, it uses straight `testing` and I'd prefer that it pre-generated with testify assert/require already in place.